server/src/routes/budgets.ts

```diff
@@ -66,4 +66,67 @@ router.post(
   },
 );
 
+/**
+ * Update an budget by id
+ *
+ * This route updates an budget by id for the authenticated user.
+ *
+ * @route PUT /budgets/:budgetId
+ * @param budgetId Id of the budget
+ * @param amount Amount of the budget
+ * @param duration Duration of the budget, either weekly, monthly, or yearly
+ * @param category Category of the budget, must be a valid category in Prisma model, or null
+ * @returns {object} An object containing the budget
+ * @throws {object} 500 - If there is a server error
+ */
+router.put(
+  '/:budgetId',
+  authenticate,
+  body('amount').isNumeric().withMessage('Amount must be a number'),
+  body('duration')
+    .isString()
+    .isIn(Object.values(BudgetDuration))
+    .withMessage('Duration must be a valid duration'),
+  body('category')
+    .optional()
+    .isString()
+    .isIn(Object.values(ExpenseCategory))
+    .withMessage('Category must be a valid category'),
+  async (req: Request, res: Response) => {
+    const { budgetId } = req.params;
+
+    try {
```

Comment:
We need to check if there were any validation errors, and return a 400 with the errors.
```typescript
    const errors = validationResult(req);
    if (!errors.isEmpty()) {
      return res.status(400).json({ errors: errors.array() });
    }
```

Reply:
fixed in latest commit, line 96